apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/AbstractTracerContext.java

```diff
@@ -31,6 +31,15 @@
      */
     void inject(ContextCarrier carrier);
 
+    /**
+     * Prepare for the cross-process propagation based on the given exit span. The given exit span should belong to the
+     * current context. How to initialize the carrier, depends on the implementation.
+     *
+     * @param carrier  to carry the context for crossing process.
+     * @param exitSpan to represent the scope of current injection.
+     */
+    void _inject(AbstractSpan exitSpan, ContextCarrier carrier);
```

Comment:
why add the leading `_`

Reply:
I was wanting people think this could be used as an open API. That is the only repo. Do you want to keep the same method name? `inject`?

Reply:
> I was wanting people think this could be used as an open API. That is the only repo. Do you want to keep the same method name? `inject`?

All methods in an `interface` are **OPEN API** IMO, there's no conventions that methods start with `_` is private or open, on the contrary, many languages use the leading `_` to indicate that the method is `private`

Reply:
From my idea, the interface is just open at the codes level, in here, to balance the noop tracing context and real tracing context. Anyway, I could change it to `inject` to more clear comment.

Reply:
Done.

Reply:
> I was wanting people think this could be used as an open API. 

Well, I think you just expressed the opposite meanings ("want people **NOT** to used as an open API"), which makes me more confused why add this method here, instead of in `TracingContext` (package private), the `ExitSpan` holds an instance of `TracingContext` instead of `AbstractTracerContext`, it should illuminate both our concerns

Reply:
Good point. Let me do that.

Reply:
Fixed, should be better now.